### PR TITLE
Fix Coverity 1494385 logically dead code.

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
Is this really worth it? IMO this is clear example of where I'd just add a comment and be with it.

**Comment by paulidale** [comment]
Not addressing the underlying problem, means it will come back when the code moves.  Fixing this now so there is no "error" to be discovered prevent this.  This happened a lot as part of 3.0 development.

Otherwise, I'd agree that a comment on the Coverity issue is sufficient.

**Comment by t8m** [comment]
I mean even adding a comment to the code would be IMO OK, but removing the code that is there for a reason, it's just not called currently seems a little bit too much to me.

**Comment by paulidale** [comment]
It seems unlikely it will ever be called.  2^n mod 5 is never zero.  Likewise, the intermediate values (e.g. 1536) mod 5 are also never zero.  This really is dead code.

**Comment by paulidale** [comment]
Ping for review?

Does anyone else think that leaving the detected "problem" there with a warning comment is a better solution than fixing the "problem"?

**Review by beldmit — APPROVED** [review]
LGTM

**Comment by paulidale** [comment]
Merged to master and 3.0 with some manual fixes.

File: crypto/bn/rsaz_exp_x2.c [hunk 1]
```diff
@@ -404,14 +405,23 @@ int RSAZ_mod_exp_x2_ifma256(BN_ULONG *out,
...
+         *      exp_bit_no = modulus_bitsize - exp_win_size
+         * However, this isn't possible because rem is { 1024, 1536, 2048 } % 5
+         * which is { 4, 1, 3 } respectively.
+         *
+         * If this assertion ever fails the fix above is easy.
+         */
+        OPENSSL_assert(rem != 0);
+
         /* Process 1-st exp window - just init result */
         BN_ULONG red_table_idx_0 = expz[exp_chunk_no + 0 * (exp_digits + 1)];
```
Comment (bernd-edlinger) [inline crypto/bn/rsaz_exp_x2.c:426]: this is not allowed in C90, declarations after a statement

**Pull request closed without merging**